### PR TITLE
Fix --no-icu tests and add CI for it

### LINKED ISSUE #4734: Add CI for `build.sh --no-icu`
If we are going to support this configuration, well, we should actually support it. This includes both marking existing tests that fail without ICU/WinGlob as exclude as well as adding the CI definition.

Right now, theres a tag exclude_noicu, that should be sufficient for marking the current failures (All of the Intl tests, an es6 test, and an es6module test currently fail).

### PR DESCRIPTION
**Pull request by jackhorton**
Fixes #4734

### CODE REVIEW

**Comment by jackhorton** [comment]
Is it reasonable to make it only for PRs? I don't expect enough churn in --no-icu situations that it requires daily builds or OS X builds.

**Comment by dilijev** [comment]
The rolling build isn't very expensive either, and this will be part of our green matrix that we want to keep green, right? When branches automatically merge it's nice to make sure nothing breaks. It's difficult to display a badge for PR builds, so better to have a badge for the rolling build instead. Daily, OSX, etc are not necessary.

File: test/es6module/rlexe.xml [hunk 1]
```diff
@@ -87,7 +87,7 @@
     <default>
       <files>module-namespace.js</files>
       <compile-flags>-ES6Module -Es6ToStringTag -args summary -endargs</compile-flags>
-      <tags>exclude_dynapogo,exclude_drt,exclude_sanitize_address</tags>
+      <tags>exclude_dynapogo,exclude_drt,exclude_sanitize_address,exclude_noicu</tags>
```
Comment (boingoing) [inline test/es6module/rlexe.xml]: Why does this test fail with noicu?

Comment (jackhorton) [reply test/es6module/rlexe.xml]: There are some Unicode names in there that get sorted -- I assumed that was related, but I didn't look into it too much. I can open a bug for it.

**Comment by obastemur** [comment]
:shipit:

**Comment by jackhorton** [comment]
@dotnet-bot test ci please

**Comment by jackhorton** [comment]
@dilijev I think I have set it up now so there are rolling and PR tasks on all relevant branches. If the wiki is to be believed, assuming `test ci please` passes, I have to merge this and just hope it worked correctly this weekend to test it?

**Comment by jackhorton** [comment]
@dotnet-bot test ci please

**Comment by jackhorton** [comment]
CI from master-ci: https://ci2.dot.net/job/Microsoft_ChakraCore/job/master-ci/job/GenPRTest/job/_no_icu_static_ubuntu_linux_debug_prtest/2/

**Comment by dilijev** [comment]
> I have to merge this and just hope it worked correctly this weekend to test it?

Not really, that's what the `master-ci` branch is for. Merge your changes to that branch, then create a PR with these changes against that branch, and if that CI passes, then your checks worked and pass.  Then you can merge this change to `master`.

**Comment by jackhorton** [comment]
Did that according to @MSLaguana's offline suggestion and it seemed to pass, so I will merge this today.

**Comment by jackhorton** [comment]
I also updated the wiki page about adding CI jobs to mention that. https://github.com/Microsoft/ChakraCore/wiki/Jenkins-CI-Checks/_compare/485011af1a4b185031c1e52ed60a4fb67299d1ba...747743df39efca9aa8f16fa7a93a76f979222ee1

**Pull request merged**